test/ethregistrar/TestEthRegistrarController.js

```diff
@@ -1,185 +1,727 @@
-
-        var nodehash = namehash.hash("newconfigname2.eth");
-        assert.equal((await ens.resolver(nodehash)), resolver.address);
-        assert.equal((await resolver.addr(nodehash)), 0);
-    });
+      assert.equal(await controller.available('newname'), false)
+    })
+
+    it('should permit new registrations with resolver and records', async () => {
+      var commitment = await controller.makeCommitment(
+        'newconfigname',
+        registrantAccount,
+        secret,
+        resolver.address,
+        [
+          resolver.interface.encodeFunctionData('setAddr(bytes32,address)', [
+            namehash.hash('newconfigname.eth'),
+            registrantAccount,
+          ]),
+          resolver.interface.encodeFunctionData('setText', [
+            namehash.hash('newconfigname.eth'),
+            'url',
+            'ethereum.com',
+          ]),
+        ],
+        false,
+        0
+      )
+      var tx = await controller.commit(commitment)
+      assert.equal(
+        await controller.commitments(commitment),
+        (await web3.eth.getBlock(tx.blockNumber)).timestamp
+      )
+
+      await evm.advanceTime((await controller.minCommitmentAge()).toNumber())
+      var balanceBefore = await web3.eth.getBalance(controller.address)
+      var tx = await controller.register(
+        'newconfigname',
+        registrantAccount,
+        secret,
+        resolver.address,
+        [
+          resolver.interface.encodeFunctionData('setAddr(bytes32,address)', [
+            namehash.hash('newconfigname.eth'),
+            registrantAccount,
+          ]),
+          resolver.interface.encodeFunctionData('setText', [
+            namehash.hash('newconfigname.eth'),
+            'url',
+            'ethereum.com',
+          ]),
+        ],
+        false,
+        0,
+        { value: 28 * DAYS }
+      )
+
+      const block = await provider.getBlock(tx.blockNumber)
+
+      await expect(tx)
+        .to.emit(controller, 'NameRegistered')
+        .withArgs(
+          'newconfigname',
+          sha3('newconfigname'),
+          registrantAccount,
+          28 * DAYS,
+          0,
+          block.timestamp + 28 * DAYS
+        )
+
+      assert.equal(
+        (await web3.eth.getBalance(controller.address)) - balanceBefore,
+        28 * DAYS
+      )
+
+      var nodehash = namehash.hash('newconfigname.eth')
+      assert.equal(await ens.resolver(nodehash), resolver.address)
+      assert.equal(await ens.owner(nodehash), nameWrapper.address)
+      assert.equal(
+        await baseRegistrar.ownerOf(sha3('newconfigname')),
+        nameWrapper.address
+      )
+      assert.equal(await resolver['addr(bytes32)'](nodehash), registrantAccount)
+      assert.equal(await resolver['text'](nodehash, 'url'), 'ethereum.com')
+      assert.equal(await nameWrapper.ownerOf(nodehash), registrantAccount)
+    })
+
+    it('should not permit new registrations with records updating a different name', async () => {
```

Comment:
You can add another test case making sure `ETHRegistrarController: Failed to set Record` (maybe pass empty  or wrong resolver address)

Reply:
Done. Added one for an incorrect resolver and an EoA. `0x0...` resolver gets caught earlier in the makeCommitment, which I also added a test for